python/feathub/processors/spark/dataframe_builder/tests/test_spark_derived_feature.py

```diff
@@ -0,0 +1,135 @@
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import shutil
+import tempfile
+import unittest
+from typing import Optional, List
+
+import pandas as pd
+
+from feathub.common import types
+from feathub.common.types import Float64, from_numpy_dtype
+from feathub.feature_tables.sources.file_system_source import FileSystemSource
+from feathub.feature_views.derived_feature_view import DerivedFeatureView
+from feathub.feature_views.feature import Feature
+from feathub.online_stores.memory_online_store import MemoryOnlineStore
+from feathub.processors.spark.spark_processor import SparkProcessor
+from feathub.processors.spark.spark_table import SparkTable
+from feathub.registries.local_registry import LocalRegistry
+from feathub.table.schema import Schema
+
+
+class SparkDerivedFeatureViewTest(unittest.TestCase):
+    def setUp(self) -> None:
+        self.registry = LocalRegistry(props={})
+        self.temp_dir = tempfile.mkdtemp()
+
+    def tearDown(self) -> None:
+        MemoryOnlineStore.get_instance().reset()
+        shutil.rmtree(self.temp_dir, ignore_errors=True)
+
+    def test_derived_feature_view(self):
+        input_data = pd.DataFrame(
+            [
+                ["Alex", 100, 100, "2022-01-01 08:01:00"],
+                ["Emma", 400, 250, "2022-01-01 08:02:00"],
+                ["Alex", 300, 200, "2022-01-02 08:03:00"],
+                ["Emma", 200, 250, "2022-01-02 08:04:00"],
+                ["Jack", 500, 500, "2022-01-03 08:05:00"],
+                ["Alex", 600, 800, "2022-01-03 08:06:00"],
+            ],
+            columns=["name", "cost", "distance", "time"],
+        )
+
+        schema = (
+            Schema.new_builder()
+            .column("name", types.String)
+            .column("cost", types.Int32)
+            .column("distance", types.Int32)
+            .column("time", types.String)
+            .build()
+        )
+
+        source = self._create_file_source(input_data, schema=schema)
+
+        f_cost_per_mile = Feature(
+            name="cost_per_mile",
+            dtype=Float64,
+            transform="CAST(cost AS DOUBLE) / CAST(distance AS DOUBLE) + 10",
+        )
+
+        features = DerivedFeatureView(
+            name="feature_view",
+            source=source,
+            features=[
+                f_cost_per_mile,
+            ],
+            keep_source_fields=False,
+        )
+
+        processor = SparkProcessor(
+            props={"processor.spark.master": "local[1]"},
+            registry=self.registry,
+        )
+
+        result_df = (
+            SparkTable(
```

Comment:
I want to double-check whether the following methods are public API.

- `SparkTable.__init__`
- `SparkDataFrameBuilder.build`
- `SparkTable.to_pandas`
- `SparkTable.execute_insert`

And if `xxxTable.execute_insert` is a public API, what is the difference between it and `xxxProcessor.materialize_features`?

Besides, I agree with it that the tests shoudl cover Feathub public APIs, but this statement works for tests in `spark/tests`. As for tests in `spark/dataframe_builder/tests`, they should verify the correctness of the public APIs of `SparkDataFrameBuilder`. Thus do you think it would be better if I modify this part of code as follows?
```python
        result_df = (
            self.dataframe_builder.build(features)
            .to_pandas()
            .sort_values(by=["name", "time"])
            .reset_index(drop=True)
        )
```